test/jdk/java/awt/MenuItem/LotsOfMenuItemsTest.java

```diff
@@ -0,0 +1,112 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+
+/*
+ * @test
+ * @bug 4175790
+ * @requires os.family == "windows"
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Win32: Running out of command ids for menu items
+ * @run main/manual LotsOfMenuItemsTest
+ */
+
+public class LotsOfMenuItemsTest extends ComponentAdapter {
+    private static final int NUM_WINDOWS = 400;
+    private static TestFrame firstFrame;
+    private static TestFrame testFrame;
+
+    public static void main(String[] args) throws Exception {
+        LotsOfMenuItemsTest obj = new LotsOfMenuItemsTest();
+        String INSTRUCTIONS = """
+                This test creates lots of frames with menu bars.
+                When it's done you will see two frames.
+                Try to select menu items from each of them.
+
+                If everything seems to work - test passed.
+                Click "Pass" button in the test harness window.
+
+                If test crashes on you - test failed.""";
+
+        PassFailJFrame.builder()
+                .title("LotsOfMenuItemsTest")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(obj::createAndShowUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private Frame createAndShowUI() {
+        firstFrame = new TestFrame("First frame", false);
+        firstFrame.addComponentListener(this);
+        return firstFrame;
+    }
+
+    @Override
+    public void componentShown(ComponentEvent e) {
+        for (int i = 1; i < NUM_WINDOWS - 1; ++i) {
+            testFrame = new TestFrame("Running(" + i + ")...");
+            testFrame.setVisible(false);
+            testFrame.dispose();
+        }
+        testFrame = new TestFrame("Last Frame");
+        testFrame.setLocation(firstFrame.getX(),
+                firstFrame.getY() + firstFrame.getHeight() + 8);
+        PassFailJFrame.addTestWindow(testFrame);
+    }
+
+    private static class TestFrame extends Frame {
+        static int n = 0;
+
+        public TestFrame(String title) {
+            this(title, true);
+        }
+
+        public TestFrame(String title, boolean visible) {
```

Comment:
Most of these comments are already resolved, except for

> `testFrame` field is no longer needed, it can be a local variable in `componentShown`.

I discussed it with Abhishek privately but I forgot to click the *Submit* button to post them.